diff --git a/picohttp/h3zero_common.c b/picohttp/h3zero_common.c
index 2ca41c42..0d85c998 100644
--- a/picohttp/h3zero_common.c
+++ b/picohttp/h3zero_common.c
@@ -1583,7 +1583,7 @@ int h3zero_process_h3_client_data(picoquic_cnx_t* cnx,
 					}
 					else
 					{
-						if (ret == 0 && ctx->no_disk == 0) {
+						if (ret == 0 && ctx->no_disk == 0 && stream_ctx->F != NULL) {
 							ret = (fwrite(bytes, 1, available_data, stream_ctx->F) > 0) ? 0 : -1;
 							if (ret != 0) {
 								picoquic_log_app_message(cnx,
